### ISSUE TITLE
Power9 guests using KUAP can lockup the host

### ISSUE THREAD

**Issue by mpe** [linuxppc/issues#292]
On power9 if a guest uses KUAP it can lead to a lockup in the host.

This happens if the host kernel doesn't have:
torvalds/linux@e9cef0189c5b or torvalds/linux@53a712bae5dd

The mechanism is that a guest running on some hardware thread goes idle with the AMR set to block user space access. A host thread also goes idle. At some point the host thread wakes up from idle, due to thread reconfiguration it may wake up on the same hardware thread that the guest went idle on. This means the host can wake up with the guest AMR value, and because of the missing commit we fail to restore a sane value.

The end result is the host kernel is running with AMR set to block user space access, but the host kernel is not KUAP aware and doesn't know to clear AMR. Eventually a user access will incorrectly be blocked by AMR, which can lead to a host lockup.

The fix is "simply" to backport one of the above commits to all power9 kernels.

**Comment by mpe** [comment]
@ajdlinux what's the status on this?

I think you have done backports to some versions?

**Comment by mpe** [comment]
This is now [CVE-2020-11669](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-11669)

**Comment by ajdlinux** [comment]
Backports for 4.14 and 4.19 have been tested, retested, tested yet again, and sent.

I started work on a backport for 4.9 which is a bit more complex however we determined that 4.9 does not require a backport, as it doesn't support KVM on POWER9 hosts and we aren't aware of other ways to hit this issue.

The writeup is at https://lists.ozlabs.org/pipermail/linuxppc-dev/2020-April/208663.html and has also been sent to oss-security.
